src/flint/FlintTypes.jl

```diff
@@ -410,7 +410,7 @@ The ring $\mathbb Z/n\mathbb Z$ for some $n$. See [`residue_ring`](@ref).
 Implementation for the modulus being a machine integer [`Int`](@ref).
 For the modulus being a [`ZZRingElem`](@ref) see [`ZZModRing`](@ref).
 """
-@attributes mutable struct zzModRing <: Ring
+@attributes mutable struct zzModRing <: ResidueRing{UInt}
```

Comment:
I would argue that this should also be `ResidueRing{ZZRingElem}`?

Reply:
Well we have
```
julia> supertypes(zzModRingElem)
(zzModRingElem, ResElem{UInt64}, RingElem, NCRingElem, SetElem, Any)
```
and these should match, I think?

Reply:
makes, sense, unfortunately.